AK/JsonValue.h

```diff
@@ -90,6 +91,13 @@ class JsonValue {
             return as_string();
         return serialized<StringBuilder>();
     }
+
+    ErrorOr<String> to_string() const
+    {
+        if (is_string())
+            return String::from_deprecated_string(as_string());
+        return String::from_deprecated_string(serialized<StringBuilder>());
+    }
```

Comment:
No. New to_string functions should propagate errors all the way, this is no improvement over the existing `to_deprecated_string()`. Temporary conversions between the old and new types should happen at the call site.

Reply:
> No. New to_string functions should propagate errors all the way, this is no improvement over the existing `to_deprecated_string()`. Temporary conversions between the old and new types should happen at the call site.

`String::from_deprecated_string` already handles errors, and getting rid of that would require actually porting all the JsonValue to String.

Reply:
> String::from_deprecated_string already handles errors

Not allocation errors though, which is ultimately the primary concern of the new string type. UTF-8 validation is a nice bonus, but secondary.

> and getting rid of that would require actually porting all of JsonValue to String.

Right, there's more work needed here than wrapping this in a `from_deprecated_string()` and calling it a day :)

Reply:
Ok, I will mark this as draft for now and port JsonValue to String separately (unless someone does this first)